src/github_services.py

```diff
@@ -0,0 +1,259 @@
+        activity_url = ISSUE_TIMELINE_URL_TEMPLATE.format(
+            org_name, repository, pr_number)
+
+        page_number = 1
+        while True:
+            logging.info('Fetching PR #%s timeline', pr_number)
+            response = requests.get(
+                activity_url,
+                params={'page': page_number, 'per_page': 100},
+                headers={
+                    'Accept': 'application/vnd.github+json',
+                    'Authorization': f'token {_TOKEN}'}
+            )
+            response.raise_for_status()
+            timeline_subset = response.json()
+
+            if len(timeline_subset) == 0:
+                break
+
+            for event in timeline_subset:
+                __process_activity(pull_request, event)
+
+            page_number += 1
+
+
+@check_token
+def create_discussion_comment(
+    org_name: str,
+    repo: str,
+    discussion_category: str,
+    discussion_title: str,
+    body: str
+) -> None:
+    """Comment in the existing GitHub discussion."""
+
+    query = """
+        query ($org_name: String!, $repository: String!) {
+            repository(owner: $org_name, name: $repository) {
+                discussionCategories(first: 10) {
+                    nodes {
+                        id
+                        name
+                        repository {
+                            id
+                            discussions(last: 10) {
+                                edges {
+                                    node {
+                                        id
+                                        title
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    """
+
+    variables = {
+        'org_name': org_name,
+        'repository': repo
+    }
+
+    response = requests.post(
+        GITHUB_GRAPHQL_URL,
+        json={'query': query, 'variables': variables},
+        headers=_get_request_headers()
+    )
+
+    data = response.json()
+
+    discussion_id = None
+    discussion_categories = (
+        data['data']['repository']['discussionCategories']['nodes'])
+
+    for category in discussion_categories:
+        if category['name'] == discussion_category:
+            discussions = category['repository']['discussions']['edges']
+            for discussion in discussions:
+                if discussion['node']['title'] == discussion_title:
+                    discussion_id = discussion['node']['id']
+                    break
+            if discussion_id is None:
+                raise Exception(
+                    f'Discussion with title {discussion_title} not found, please create a '
+                    'discussion with that title.')
+            break
+
+    if discussion_id is None:
+        raise Exception(f'{discussion_category} category is missing in GitHub Discussion.')
+
+    query = """
```

Comment:
Maybe you could add a comment explaining what's going on here a bit, with a reference to the language it's written in?

Reply:
added, please let me know if that works